### PR TITLE
feat: add functions to load EIA and EPA data from blob storage

### PR DESCRIPTION
**Pull request by danielolsen**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Add functions to load EIA and EPA data from blob storage (or from a local path).

### What the code is doing
- We add blob storage paths in **const.py**.
- `get_eia_form_860` takes a path and loads a data frame, with some simple filtering based on state abbreviations.
- `get_epa_needs` takes a path and loads a data frame, with some simple filtering based on full state names.
- `get_epa_ampd` takes a path to a folder or URL, loads data frames for every relevant filepath within that folder/URL, and aggregates them. The `path` parameter is automatically parsed to determine whether it's a URL or a local path.

### Testing
Tested manually, all files load successfully. ~except for the NEEDS file, which seems to be because of an encoding error in the file in our blob storage (ANSI instead of UTF-8):~ EDIT: this encoding error has been fixed.
```python
>>> from prereise.gather.griddata.hifld import const
>>> from prereise.gather.griddata.hifld.data_access.load import get_eia_form_860
>>> from prereise.gather.griddata.hifld.data_access.load import get_epa_ampd
>>> from prereise.gather.griddata.hifld.data_access.load import get_epa_needs
>>> # Blob storage path, no trailing slash
>>> epa_ampd = get_epa_ampd("https://besciences.blob.core.windows.net/datasets/EPA_AMPD")
>>> # Blob storage path, trailing slash
>>> epa_ampd = get_epa_ampd("https://besciences.blob.core.windows.net/datasets/EPA_AMPD/")
>>> # Local path, no trailing slash
>>> epa_ampd = get_epa_ampd(r"C:\Users\DanielOlsen\Breakthrough Energy Dropbox\BreakthroughEnergySciences\SourceData\EPA_AMPD")
>>> # Local path, trailing slash
>>> epa_ampd = get_epa_ampd(r"C:\Users\DanielOlsen\Breakthrough Energy Dropbox\BreakthroughEnergySciences\SourceData\EPA_AMPD\\")
>>> eia_form_860 = get_eia_form_860(const.blob_paths["eia_form860_2019"])
>>> epa_needs = get_epa_needs(const.blob_paths["epa_needs"])
```

### Time estimate
15 minutes.

### CODE REVIEW

**Comment by rouille** [comment]
Should we update the **ATTRIBUTION** file in this PR?

**Comment by danielolsen** [comment]
> Should we update the **ATTRIBUTION** file in this PR?

Done.

File: prereise/gather/griddata/hifld/data_access/load.py [hunk 1]
```diff
@@ -10,6 +11,65 @@
...
+    try:
+        result = urlparse(path)
+        if result.scheme in {"http", "ftp", "s3", "file"}:
+            path_sep = "/"
+        else:
+            path_sep = os.path.sep
+    except Exception:
+        raise ValueError(f"Could not interpret path {path}")
+    # Trim trailing slashes as necessary to ensure that join works
+    if path[-1] in {"/", "\\"}:
```
Comment (jon-hagg) [inline prereise/gather/griddata/hifld/data_access/load.py:44]: Could also use `path = path.rstrip("/\\")`

Comment (danielolsen) [reply prereise/gather/griddata/hifld/data_access/load.py]: Nice! Done.

**Review by rouille — APPROVED** [review]
Thanks

**Pull request merged**